lib/workers/repository/update/pr/labels.ts

```diff
@@ -2,15 +2,29 @@ import is from '@sindresorhus/is';
 import { dequal } from 'dequal';
 import type { RenovateConfig } from '../../../../config/types';
 import { logger } from '../../../../logger';
+import { platform } from '../../../../modules/platform';
 import * as template from '../../../../util/template';
 
+/**
+ * Filter labels that exceed maximum char limit based on platform limits
+ */
+function trimLabel(label: string, limit: number): string {
+  if (label.length > limit) {
+    return label.slice(0, limit).trim();
+  }
+
+  return label.trim();
+}
+
 export function prepareLabels(config: RenovateConfig): string[] {
+  const labelCharLimit = platform.labelCharLimit ?? 50;
```

Comment:
remove from platforms GitHub and gitea, as you use same default 

add a note to the docs that labels are generally truncated to 50 chars for all platforms but gitlab which has higher limits